### PR TITLE
identical values from interfaces

### PR DESCRIPTION
**Pull request by xiangli313**
https://mrc-ide.myjetbrains.com/youtrack/issue/VIMC-5232

### CODE REVIEW

File: R/impact_central.R [hunk 1]
```diff
@@ -431,24 +431,22 @@ impact_by_year_of_vaccination_activity_type <- function(
   baseline_burden, focal_burden, fvps, vaccination_years) {
```
Comment (r-ash) [inline R/impact_central.R:420]: The docs for this say "this method produces multiple, activity-specific impact ratios" this is no longer true, we should update this. It only work for 1 activity type.

Comment (r-ash) [reply R/impact_central.R:420]: More generally, the idea of these functions was that they would match the format from the paper. After these changes are we still satisfied that they match what the paper describes? @kgaythorpe

File: R/impact_central.R [hunk 2]
```diff
@@ -496,7 +493,7 @@ impact_by_year_of_vaccination_activity_type <- function(
     dplyr::inner_join(impact_ratio, by = c("country", "activity_type")) %>%
     dplyr::mutate(impact = impact_ratio * fvps) %>%
     dplyr::select(country, vaccine, activity_type, year, burden_outcome,
-                  impact) %>%
+                  impact, fvps) %>%
```
Comment (r-ash) [inline R/impact_central.R]: Where are these used? Are the fvps needed by some report?

**Review by r-ash — CHANGES_REQUESTED** [review]
There are a couple of test failures - Xiang can you review they are changing things how you expect? 

```
── Failure (test-db-impact.R:392:3): impact by YOV activity type: different impact & fvp ──
nrow(impact) (`actual`) not equal to 0 (`expected`).

  `actual`: 5
`expected`: 0
```
Previously returned 0 because we had different activity type for focal/burden & fvps but the code no longer cares about that (it ignores activity_type in focal & burden data) - is this the behaviour we want? Are we just expecting people to use the correct focal & burden data for the fvps? Links into comments about the public facing interface.

```
── Failure (test-unit-impact-meta.R:47:3): test replace_burden_outcomes ────────
`replace_burden_outcome(burden_outcomes, "cases;dalys")` did not throw the expected error.
Backtrace:
    █
 1. └─testthat::expect_error(...) test-unit-impact-meta.R:47:2
 2.   └─testthat:::expect_condition_matching(...)
```
replace_burden_outcome has been updated - is this test now obsolete?

 Xiang can you review the test changes I have made here too to make sure they match what you would expect? @kgaythorpe could you review the science changes please?

File: R/impact_central.R (hunk 2 above)
Comment (xiangli313) [reply R/impact_central.R]: We need impact_ratios and fvps in our reports for funders. I picked up fvps, but maybe impact_ratio is a better choice. fvps can be attached at a later stage, but impact ratio will be lost if not in the output.

File: tests/testthat/test-db-impact.R [hunk 3]
```diff
@@ -376,35 +376,42 @@ test_that("impact by year of vaccination activity type: only routine", {
...
+#   ## now we swtiched to fvps for determining which activity_type to work on
+#   ## hence expect_equal(nrow(impact), 0) is no longer true
+#   ## I guess this test is no longer useful, but good to keep here as a record of what changes we've made
+#   baseline <- impact_test_data_baseline[
+#     impact_test_data_baseline$activity_type == "routine", ]
+#   focal <- impact_test_data_focal[
+#     impact_test_data_focal$activity_type == "routine", ]
+#   fvps <- fvp_test_data_15[fvp_test_data_15$activity_type == "campaign", ]
+#   impact <- impact_by_year_of_vaccination_activity_type(baseline, focal,
+#                                                         fvps, 2000:2030)
```
Comment (r-ash) [inline tests/testthat/test-db-impact.R]: `impact_by_year_of_vaccination_activity_type` is going to read your `baseline` and `focal` as for whatever activity `fvp` data is. i.e. if you only pass `routine` burden data but `campaign` fvp data the function now just uses that `routine` burden data in its calculations anyway which I think is misleading.

Comment (r-ash) [reply tests/testthat/test-db-impact.R]: Ah or if we just ignore `activity_type` column of burden data then we should remove that column explicitly in the test to be clear. 

I also think if this test is obsolete now we should remove it, not much value in old tests sitting around going stale

**Review by r-ash — CHANGES_REQUESTED** [review]

**Review by r-ash — APPROVED** [review]

**Pull request merged**